### DISCUSSION TITLE
foreign key constraint fails in mirthdb

### DISCUSSION THREAD

**Discussion by sproket** [nextgenhealthcare/connect#5588] [Q&A]
I see this occasionally in prod.

Cannot add or update a child row: a foreign key constraint fails (`mirthdb`.`D_MCM104`, CONSTRAINT `D_MCM104_FKEY` FOREIGN KEY (`MESSAGE_ID`, `METADATA_ID`) REFERENCES `D_MM104` (`MESSAGE_ID`, `ID`) ON DELETE CASCADE)

Any idea the cause?

**Comment by jonbartels** [comment]
Where do you see this? In the UI or in `mirth.log`?

Are there additional details in the error message like a stack trace?

What database are you using?

The "104" refers to LOCAL channel ID 104. Somehow you're getting an FK conflict. 

https://github.com/kayyagari/connect/blob/05814e97651f28fd323a7c0df5be01614a0515f5/mc-integ-tests/mc-db-tables.png

**Reply by sproket** [reply]
Thanks. It's appearing in the mirth logs and in the dashboard logs. We're using mariadb.   We have a pre-prod environment which duplicates everything.  We'll probably drop the prod and restore from pre-prod.  AND we'll have to check prod file systems.

**Reply by jonbartels** [reply]
d_mm is a connector message (so data for a source connector or destination connector). 
d_mcm is message metadata, anything in the metadata table in the channel summary tab is stored here.

For that error to happen Mirth would have to be inserting a row into d_mcm before the corresponding row in d_mm exists. That shouldn't be possible since the code will have to insert them in order for all databases.

**Comment by wi4seo** [comment]
Hello,

FREE Registration for Mirth Unconference 2023 is Now Opened!

To know more, please visit the event page and register yourself: https://wi4.org/mirth-connect-unconference-feb-2023/

Stay Tuned for More Updates!​

​Thanks
